.travis.yml

```diff
@@ -21,12 +21,13 @@ addons:
 env:
     global:
         # Set defaults to avoid repeating in most cases
+        - ASTROPY_USE_SYSTEM_PYTEST=1
```

Comment:
@astrofrog , now that we're sure it is compatible, I have no objection (though personally, I rather have the whole test suite use the same pytest for simplicity). But earlier in this PR's history, enforcing pytest 3 for all is necessary to ensure complete compatibility. Otherwise, we wouldn't have detected specific failures in Python 2, coverage, etc.